Add c++ keywords

FireChickenProductivity opened the pull request.

chdoc commented:
> If you have time, could you let me know what you think of the spacing and what should go in the unprefixed list?

I don't really have any strong opinions on the spacing. As it comes to the list of prefixed keywords, I would include those keywords that are verbally distinctive, commonly used, don't have any other commands inserting them yet, and are not accompanied by braces or other structure elements. For me, I suppose the most important ones would be `static`, `constexpr`, and `const`. `namespace` should probably be a snippet. The word that I am I'm sure about is `auto`. Personally, I have added
```
var auto <user.text>: 
    insert("auto ")
    user.code_private_variable_formatter(text)
    insert(" = ")
```
for auto-typed variable declarations, so I wouldn't include it in the unprefixed list.

FireChickenProductivity commented:
> > If you have time, could you let me know what you think of the spacing and what should go in the unprefixed list?
> 
> I don't really have any strong opinions on the spacing. As it comes to the list of prefixed keywords, I would include those keywords that are verbally distinctive, commonly used, don't have any other commands inserting them yet, and are not accompanied by braces or other structure elements. For me, I suppose the most important ones would be `static`, `constexpr`, and `const`. `namespace` should probably be a snippet. The word that I am I'm sure about is `auto`. Personally, I have added
> 
> ```
> var auto <user.text>: 
>     insert("auto ")
>     user.code_private_variable_formatter(text)
>     insert(" = ")
> ```
> 
> for auto-typed variable declarations, so I wouldn't include it in the unprefixed list.

Thank you for your feedback. I added the keywords you suggested and `explicit` and `virtual`.

AndreasArvidsson commented on lang/cpp/code_keyword.talon-list:
```diff
@@ -0,0 +1,109 @@
+-
+asm: asm
+aye es em: asm
+align as: "alignas"
```
```suggestion
align as: alignas
```
